packages/stg-evm-v2/ts-src/utils/helpers.ts

```diff
@@ -21,3 +26,136 @@ export const appendDependencies =
+        str = str.replace(placeholder, address)
+    }
+
+    return str
+}
+
+/**
+ * Helper function that saves deployment information to the deployments folder
+ * @returns the saved deployment
+ */
+export const saveDeployment = async ({
+    hre,
+    deploymentName,
+    deploymentContract,
+    abi,
+    creationBytecode,
+    deployedBytecode,
+    libraries,
+    args = [],
+    metadata,
+}: {
+    hre: HardhatRuntimeEnvironment
+    deploymentName: string
+    deploymentContract: Contract
+    abi: any
+    creationBytecode: string
+    deployedBytecode: string
+    libraries?: Libraries
+    args?: any[]
+    metadata: string
+}) => {
+    const deployment: Deployment = {
+        address: deploymentContract.address,
+        abi: abi as any,
+        // TODO transationHash and receipt are nice to have but not necessary (for verification)
+        transactionHash: deploymentContract.deployTransaction.hash,
+        receipt: await deploymentContract.deployTransaction.wait(),
+        args,
+        bytecode: creationBytecode,
+        deployedBytecode: deployedBytecode,
+        libraries: libraries ?? {},
+        metadata,
+    }
+
+    await hre.deployments.save(deploymentName, deployment)
+
+    return deployment
+}
+
+/**
+ * Helper function that deploys a contract and saves the deployment information
+ * @returns the saved deployment
+ */
+export const deploy = async ({
+    hre,
+    contractName,
+    deploymentName,
+    overrides,
+    abi,
+    creationBytecode,
+    signer,
+    logger,
+    libraries,
+    args = [],
+    metadata,
+}: {
+    hre: HardhatRuntimeEnvironment
+    contractName: string
+    deploymentName: string
+    overrides: object
+    abi: any
+    creationBytecode: string
+    signer: SignerWithAddress
+    logger: Logger
+    libraries?: Libraries
+    args?: any[]
+    metadata: string
+}) => {
+    const existingDeployment = await hre.deployments.getOrNull(deploymentName)
+
+    if (existingDeployment?.bytecode !== creationBytecode) {
+        logger.info(`Deploying contract ${contractName} as ${deploymentName}`)
+
+        const contractFactory = new ContractFactory(abi, creationBytecode, signer)
+
+        let contract: Contract
+        if (args) {
+            contract = await contractFactory.deploy(...args, overrides)
+        } else {
+            contract = await contractFactory.deploy(overrides)
+        }
+
+        await contract.deployed()
+
+        const contractDeployment = await saveDeployment({
+            hre,
+            deploymentName,
+            deploymentContract: contract,
+            abi,
+            creationBytecode,
+            deployedBytecode: await hre.ethers.provider.getCode(contract.address),
+            libraries,
+            args,
+            metadata,
+        })
+
+        logger.info(`${deploymentName} is deployed: ${contract.address}`) // TODO weird issue where this does not print for last contract deployed (proxy in this case)
```

Comment:
See todo, I am still investigating this but this line only seems to print for the first 2 deployments (SignatureChecker and Impl), it doesn't print when the proxy is deployed. What's strange is that `console.log` seems to work just fine and the Proxy is definitely being deployed because the deployment file is saved and I can see the contract on the block explorer